### DISCUSSION TITLE
Python Script Issue

### DISCUSSION THREAD

**Discussion by jab0707** [SCIInstitute/SCIRun#2226] [Q&A] [answered]
I am getting a weird python error. Below is the code in question in an interfacewithpython module

anchor, point1, and point2 are defined as input fields. I get all the way to the print line, which works and prints the correct values. Then on the next line, I get "firstVectorPt" is not defined.

cent = anchor['node']
firstVectorPt = point1['node']
secondVectorPt = point2['node']
print(firstVectorPt)
v1 = [firstVectorPt[ix] - cent[ix] for ix in range(3)]

[testClipPlane.srn5.zip](https://github.com/SCIInstitute/SCIRun/files/5922091/testClipPlane.srn5.zip)

**Comment by jab0707** [comment]
@dcwhite please advise. What am I doing wrong here

**Comment by jab0707** [comment]
I tried writing it out as a formal for loop and still get definition errors:
 The following says that v1 is undefined.

v1 = [0,0,0]
for ix in range(3):
	print(firstVectorPt[ix]-cent[ix])
	v1[ix] = firstVectorPt[ix]-cent[ix]

**Reply by jab0707** [reply]
I then tried it the non-loop way, and I get a 'v1 is not defined'

v1 = [0,0,0]
v1[0] = firstVectorPt[0] - cent[0]
v1[1] = firstVectorPt[2] - cent[1]
v1[2] = firstVectorPt[1] - cent[2]

**Comment by dcwhite** [comment, marked as answer]
You've used `v1` and `v2` as "output variable names" so they can't be used as intermediate variables or normal variables--IWP will interpret a `v1 = ...` line as special code to send the value from Python to C++. Change one of the sets of names, either in the script part or the list of output variable names.

This usability problem will be improved with #2112, when we rewrite the input/output IWP code as actual functions.

**Reply by jab0707** [reply]
I replaced the code with the following, where outV1 and outV2 are the optput names for matrix outputs 1 and 2. I still get a 'firstVectorPt' not defined error

cent = anchor['node']
firstVectorPt = point1['node']
secondVectorPt = point2['node']

v1 = [firstVectorPt[ix] - cent[ix] for ix in range(3)]

v2 = [secondVectorPt[ix] - cent[ix] for ix in range(3)]

outV1 = v1
outV2 = v2

**Reply by dcwhite** [reply]
This is the same problem, the input/output name variables can not be used like normal variables, which is what prompted the linked issue. You need a "regular" variable to store the I/O variables first, before you can do things like index into them as a map. So this would work:
```
a = point1
firstVectorPt = a['node']
b = point2
secondVectorPt = b['node']
```

**Reply by dcwhite** [reply]
With the functional approach in #2112, the above could look like
```
firstVectorPt = get_fieldinput(0)['node']
secondVectorPt = get_fieldinput(1)['node']
```
